pkg/sdk/testint/tables_integration_test.go

```diff
@@ -0,0 +1,949 @@
+		require.NoError(t, err)
+		t.Cleanup(cleanupTableProvider(id))
+
+		alterRequest := sdk.NewAlterTableRequest(id).
+			WithSearchOptimizationAction(sdk.NewTableSearchOptimizationActionRequest().WithAddSearchOptimizationOn([]string{"SUBSTRING(*)", "GEO(*)"}))
+
+		err = client.Tables.Alter(ctx, alterRequest)
+		require.NoError(t, err)
+	})
+
+	// TODO: try to check more sets (ddl collation, max data extension time in days, etc.)
+	t.Run("set: with complete options", func(t *testing.T) {
+		name := random.String()
+		id := sdk.NewSchemaObjectIdentifier(database.Name, schema.Name, name)
+		comment := random.String()
+		columns := []sdk.TableColumnRequest{
+			*sdk.NewTableColumnRequest("COLUMN_1", sdk.DataTypeVARCHAR),
+			*sdk.NewTableColumnRequest("COLUMN_2", sdk.DataTypeVARCHAR),
+		}
+
+		err := client.Tables.Create(ctx, sdk.NewCreateTableRequest(id, columns))
+		require.NoError(t, err)
+		t.Cleanup(cleanupTableProvider(id))
+
+		stageFileFormats := []sdk.StageFileFormatRequest{
+			{
+				Type: sdk.Pointer(sdk.FileFormatTypeCSV),
+			},
+		}
+		stageCopyOptions := []sdk.StageCopyOptionsRequest{
+			{
+				OnError: sdk.NewStageCopyOnErrorOptionsRequest().WithSkipFile(),
+			},
+		}
+		alterRequest := sdk.NewAlterTableRequest(id).
+			WithSet(sdk.NewTableSetRequest().
+				WithEnableSchemaEvolution(sdk.Bool(true)).
+				WithStageFileFormat(stageFileFormats).
+				WithStageCopyOptions(stageCopyOptions).
+				WithDataRetentionTimeInDays(sdk.Int(30)).
+				WithMaxDataExtensionTimeInDays(sdk.Int(90)).
+				WithChangeTracking(sdk.Bool(false)).
+				WithDefaultDDLCollation(sdk.String("us")).
+				WithComment(&comment))
+
+		err = client.Tables.Alter(ctx, alterRequest)
+		require.NoError(t, err)
+		table, err := client.Tables.ShowByID(ctx, id)
+		require.NoError(t, err)
+
+		assert.Equal(t, table.Comment, comment)
+		assert.Equal(t, table.RetentionTime, 30)
+		assert.Equal(t, table.ChangeTracking, false)
+		assert.Equal(t, table.EnableSchemaEvolution, true)
+	})
+
+	t.Run("drop table", func(t *testing.T) {
+		table, tableCleanup := createTable(t, client, database, schema)
+		err := client.Tables.Drop(ctx, sdk.NewDropTableRequest(table.ID()).WithIfExists(sdk.Bool(true)))
+		if err != nil {
+			t.Cleanup(tableCleanup)
+		}
+		require.NoError(t, err)
+
+		_, err = client.Tables.ShowByID(ctx, table.ID())
+		require.ErrorIs(t, err, collections.ErrObjectNotFound)
+	})
+
+	t.Run("show tables", func(t *testing.T) {
+		table, tableCleanup := createTable(t, client, database, schema)
+		t.Cleanup(tableCleanup)
+		table2, table2Cleanup := createTable(t, client, database, schema)
+		t.Cleanup(table2Cleanup)
+
+		tables, err := client.Tables.Show(ctx, sdk.NewShowTableRequest())
+		require.NoError(t, err)
+
+		tableIds := make([]sdk.SchemaObjectIdentifier, len(tables))
+		for i, table := range tables {
+			tableIds[i] = table.ID()
+		}
+		assert.Contains(t, tableIds, table.ID())
+		assert.Contains(t, tableIds, table2.ID())
+
```

Comment:
maybe also call the custom assert function at the top of the file to check all of the fields ?

Reply:
Done